### PR TITLE
disallow retryInterval when fail over is enabled

### PR DESCRIPTION
**Pull request by njr-11**
It should not be allowed to configure retryInterval (which is single-server only and interferes with fail over) when missed task threshold is configured to enable fail over.  Reject this combination and update tests accordingly.

### CODE REVIEW

**Comment by njr-11** [comment]
#build

**Comment by njr-11** [comment]
#run-libby-bot
#build

File: dev/com.ibm.ws.concurrent.persistent/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -41,7 +41,7 @@ jndiName=JNDI name
 jndiName.desc=JNDI name.
 
 missedTaskThreshold=Missed task threshold for fail over
-missedTaskThreshold.desc=Amount of time beyond the expected start of a task execution to reserve for running it. Other members are prevented from running the task before the expiration of this interval. If the interval elapses without successful execution of the task, then the task execution is considered missed, enabling another member to attempt to run it. This enables fail over.
+missedTaskThreshold.desc=missedTaskThreshold.desc=Amount of time beyond the expected start of a task execution to reserve for running it. Other members are prevented from running the task before the expiration of this interval. If the interval elapses without successful execution of the task or the task rolls back, then the task execution is considered missed, enabling another member to attempt to run it. This enables fail over.
```
Comment (dmuelle) [inline dev/com.ibm.ws.concurrent.persistent/resources/OSGI-INF/l10n/metatype.properties]: looks like there's a typo at the start of the line- desc code appears twice?

The final sentence- `This enables failover`
What, specifically, enables failover? Need to specify - is it the process described in the previous sentence? Or missedTaskThreshold?

File: dev/com.ibm.ws.concurrent.persistent/resources/com/ibm/ws/concurrent/persistent/resources/CWWKCMessages.nlsprops [hunk 2]
```diff
@@ -55,9 +55,9 @@ CWWKC1520.out.of.range=CWWKC1520E: Configured value {0} for {1} is not within th
 CWWKC1520.out.of.range.explanation=A value that is outside of the allowed range is configured for the specified property.
 CWWKC1520.out.of.range.useraction=Configure a value within the allowed range.
 
-CWWKC1521.less.than.min=CWWKC1521E: Configured value {0} for {1} cannot be less than the configured value for {2}, which is {3}.
-CWWKC1521.less.than.min.explanation=Configuring a larger retryInterval encourages other servers to retry the failed task instead of the server on which it failed.
-CWWKC1521.less.than.min.useraction=Configure the retryInterval property so that it is larger than the missedTaskThreshold.
+CWWKC1521.not.compatible=CWWKC1521E: Configuration attribute {0} is not valid when {1} is enabled.
```
Comment (dmuelle) [inline dev/com.ibm.ws.concurrent.persistent/resources/com/ibm/ws/concurrent/persistent/resources/CWWKCMessages.nlsprops]: **change**
Configuration attribute {0} is not valid when {1} is enabled
**to**
The {0} configuration attribute is not valid when the {1} configuration attribute is enabled.

File: dev/com.ibm.ws.concurrent.persistent/resources/com/ibm/ws/concurrent/persistent/resources/CWWKCMessages.nlsprops [hunk 3]
```diff
@@ -55,9 +55,9 @@ CWWKC1520.out.of.range=CWWKC1520E: Configured value {0} for {1} is not within th
...
+CWWKC1521.not.compatible.explanation=The specified configuration attributes are not compatible.
+CWWKC1521.not.compatible.useraction=Update configuration to remove one of the configuration attributes.
```
Comment (dmuelle) [inline dev/com.ibm.ws.concurrent.persistent/resources/com/ibm/ws/concurrent/persistent/resources/CWWKCMessages.nlsprops]: **change**
Update configuration to remove one of the configuration attributes.
**to**
Update the configuration to remove one of the configuration attributes.

File: dev/com.ibm.ws.concurrent.persistent/resources/OSGI-INF/l10n/metatype.properties (hunk 1 above)
Comment (njr-11) [reply dev/com.ibm.ws.concurrent.persistent/resources/OSGI-INF/l10n/metatype.properties]: Thanks for catching this.  I attempted to fix this issue and the other issues that you found under
https://github.com/OpenLiberty/open-liberty/pull/10521/commits/726da70bbb1c8c0f3ba51a7f7f9a16e02de916b9
Please review and approve or let me know if further changes are needed.

File: dev/com.ibm.ws.concurrent.persistent/resources/OSGI-INF/l10n/metatype.properties [hunk 4]
```diff
@@ -41,7 +41,7 @@ jndiName=JNDI name
 jndiName.desc=JNDI name.
 
 missedTaskThreshold=Missed task threshold for fail over
-missedTaskThreshold.desc=Amount of time beyond the expected start of a task execution to reserve for running it. Other members are prevented from running the task before the expiration of this interval. If the interval elapses without successful execution of the task, then the task execution is considered missed, enabling another member to attempt to run it. This enables fail over.
+missedTaskThreshold.desc=Amount of time beyond the expected start of a task execution to reserve for running it. Other members are prevented from running the task before the expiration of this interval. If the interval elapses without successful execution of the task or the task rolls back, then the task execution is considered missed, enabling another member to attempt to run it. Missed task threshold values within the supported range of 100 seconds to 9000 seconds (2.5 hours) enable failover.
```
Comment (dmuelle) [inline dev/com.ibm.ws.concurrent.persistent/resources/OSGI-INF/l10n/metatype.properties]: **change**
Amount of time beyond the expected start of a task execution to reserve for running it.
**to**
The amount of time after the expected start of a task execution that is reserved for running the task.

Comment (njr-11) [reply dev/com.ibm.ws.concurrent.persistent/resources/OSGI-INF/l10n/metatype.properties]: The fix for the above has been made exactly as suggested under https://github.com/OpenLiberty/open-liberty/pull/10521/commits/9d01e544ea48cebdf0ae90ef0787d6af15ef4e25

**Comment by njr-11** [comment]
#build
#run-libby-bot

**Comment by njr-11** [comment]
#build

File: dev/com.ibm.ws.concurrent.persistent/resources/OSGI-INF/l10n/metatype.properties [hunk 5]
```diff
@@ -41,7 +41,7 @@ jndiName=JNDI name
 jndiName.desc=JNDI name.
 
 missedTaskThreshold=Missed task threshold for fail over
-missedTaskThreshold.desc=Amount of time beyond the expected start of a task execution to reserve for running it. Other members are prevented from running the task before the expiration of this interval. If the interval elapses without successful execution of the task, then the task execution is considered missed, enabling another member to attempt to run it. This enables fail over.
+missedTaskThreshold.desc=The amount of time after the expected start of a task execution that is reserved for running the task. Other members are prevented from running the task before the expiration of this interval. If the interval elapses without successful execution of the task or the task rolls back, then the task execution is considered missed, enabling another member to attempt to run it. Missed task threshold values within the supported range of 100 seconds to 9000 seconds (2.5 hours) enable failover.
```
Comment (dmuelle) [inline dev/com.ibm.ws.concurrent.persistent/resources/OSGI-INF/l10n/metatype.properties]: **suggested revision (removes "execution")

The amount of time after the expected start of a task  that is reserved for running the task. Other members are prevented from running the task before the expiration of this interval. If the interval elapses without successful completion of the task, or the task rolls back, then the task completion is considered missed, enabling another member to attempt to run it. Missed task threshold values within the supported range of 100 seconds to 9000 seconds (2.5 hours) enable failover.

Comment (njr-11) [reply dev/com.ibm.ws.concurrent.persistent/resources/OSGI-INF/l10n/metatype.properties]: Updated here:
https://github.com/OpenLiberty/open-liberty/pull/10521/commits/f40a335f74e6e1ae21c568ba8343d1a5d4ffefc2

**Comment by njr-11** [comment]
#build
#run-libby-bot

**Comment by njr-11** [comment]
An additional commit was added due to a merge conflict. It did not impact any of the message files. A new build is required though:
#build
#run-libby-bot

**Comment by andrea-pichler** [comment]
Completed the requested L2 message review

**Comment by njr-11** [comment]
#build
#run-libby-bot

**Comment by njr-11** [comment]
#build

**Comment by njr-11** [comment]
Build has 1 failure, which is a duplicate of a pre-existing issue

**Pull request merged**